### ISSUE TITLE
'verify_credentials' and 'folders' fail for Client instance

### ISSUE THREAD

**Issue by dbreunig** [stve/instapaper#5]
Error thrown:

```
/Users/dbreunig/.rvm/gems/ruby-2.1.2/gems/http-0.7.4/lib/http/mime_type.rb:36:in `[]': Unknown MIME type: text/html (HTTP::Error)
    from /Users/dbreunig/.rvm/gems/ruby-2.1.2/gems/http-0.7.4/lib/http/response.rb:100:in `parse'
    from /Users/dbreunig/.rvm/gems/ruby-2.1.2/gems/instapaper-1.0.0.pre1/lib/instapaper/http/request.rb:57:in `fail_if_error_unparseable_response'
    from /Users/dbreunig/.rvm/gems/ruby-2.1.2/gems/instapaper-1.0.0.pre1/lib/instapaper/http/request.rb:47:in `fail_if_error'
    from /Users/dbreunig/.rvm/gems/ruby-2.1.2/gems/instapaper-1.0.0.pre1/lib/instapaper/http/request.rb:42:in `perform_request'
    from /Users/dbreunig/.rvm/gems/ruby-2.1.2/gems/instapaper-1.0.0.pre1/lib/instapaper/http/request.rb:32:in `perform'
    from /Users/dbreunig/.rvm/gems/ruby-2.1.2/gems/instapaper-1.0.0.pre1/lib/instapaper/http/utils.rb:49:in `perform_request'
    from /Users/dbreunig/.rvm/gems/ruby-2.1.2/gems/instapaper-1.0.0.pre1/lib/instapaper/http/utils.rb:37:in `perform_request_with_object'
    from /Users/dbreunig/.rvm/gems/ruby-2.1.2/gems/instapaper-1.0.0.pre1/lib/instapaper/http/utils.rb:29:in `perform_post_with_object'
    from /Users/dbreunig/.rvm/gems/ruby-2.1.2/gems/instapaper-1.0.0.pre1/lib/instapaper/api/accounts.rb:9:in `verify_credentials'
    from scratch.rb:19:in `<main>'
```

Here is the code making the call:

```
credentials = {
  consumer_key: KEY,
  consumer_secret: SECRET,
  oauth_key: token.oauth_token,
  oauth_token_secret: token.oauth_token_secret
}

client = Instapaper::Client.new(credentials)
client.verify_credentials
```

**Comment by stve** [comment]
Looks like you may have invalid credentials. Where are you getting the token/secret? I just ran this example: https://github.com/stve/instapaper/blob/13b278932b1bc2065dfc4c9c2fd8318fe3054954/examples/authorization_flow.rb and didn't encounter an error. I'll improve the error handling for this scenario

**Comment by stve** [comment]
@dbreunig I just released `1.0.0.pre2` which barring any issues or setbacks, will be released as `1.0.0` shortly. I didn't hear back from Instapaper about the issues I was having with their API, but have been able to figure out workarounds for now.

**Comment by Stanleyyork** [comment]
Similarly, I updated to the 1.0.0pre2 gem and used the example you listed above, and got this error:

/Users/stanleystevens/.rvm/gems/ruby-2.2.1/gems/http-0.9.8/lib/http/headers.rb:51:in `to_a': undefined method `each' for "InstapaperRubyGem/1.0.0.pre2":String (NoMethodError)
	from /Users/stanleystevens/.rvm/gems/ruby-2.2.1/gems/http-0.9.8/lib/http/headers.rb:51:in `Array'
	from /Users/stanleystevens/.rvm/gems/ruby-2.2.1/gems/http-0.9.8/lib/http/headers.rb:51:in `add'
	from /Users/stanleystevens/.rvm/gems/ruby-2.2.1/gems/http-0.9.8/lib/http/headers.rb:182:in `block in coerce'
	from /Users/stanleystevens/.rvm/gems/ruby-2.2.1/gems/http-0.9.8/lib/http/headers.rb:182:in `each'
	from /Users/stanleystevens/.rvm/gems/ruby-2.2.1/gems/http-0.9.8/lib/http/headers.rb:182:in `coerce'
	from /Users/stanleystevens/.rvm/gems/ruby-2.2.1/gems/http-0.9.8/lib/http/headers.rb:154:in `merge!'
	from /Users/stanleystevens/.rvm/gems/ruby-2.2.1/gems/http-0.9.8/lib/http/headers.rb:162:in `block in merge'
	from /Users/stanleystevens/.rvm/gems/ruby-2.2.1/gems/http-0.9.8/lib/http/headers.rb:162:in `tap'
	from /Users/stanleystevens/.rvm/gems/ruby-2.2.1/gems/http-0.9.8/lib/http/headers.rb:162:in `merge'
	from /Users/stanleystevens/.rvm/gems/ruby-2.2.1/gems/http-0.9.8/lib/http/options.rb:59:in `block in <class:Options>'
	from /Users/stanleystevens/.rvm/gems/ruby-2.2.1/gems/http-0.9.8/lib/http/options.rb:34:in `instance_exec'
	from /Users/stanleystevens/.rvm/gems/ruby-2.2.1/gems/http-0.9.8/lib/http/options.rb:34:in `block (2 levels) in def_option'
	from /Users/stanleystevens/.rvm/gems/ruby-2.2.1/gems/http-0.9.8/lib/http/options.rb:128:in `dup'
	from /Users/stanleystevens/.rvm/gems/ruby-2.2.1/gems/http-0.9.8/lib/http/options.rb:34:in `block in def_option'
	from /Users/stanleystevens/.rvm/gems/ruby-2.2.1/gems/http-0.9.8/lib/http/chainable.rb:176:in `headers'
	from /Users/stanleystevens/.rvm/gems/ruby-2.2.1/gems/instapaper-1.0.0.pre2/lib/instapaper/http/request.rb:41:in `perform_request'
	from /Users/stanleystevens/.rvm/gems/ruby-2.2.1/gems/instapaper-1.0.0.pre2/lib/instapaper/http/request.rb:32:in `perform'
	from /Users/stanleystevens/.rvm/gems/ruby-2.2.1/gems/instapaper-1.0.0.pre2/lib/instapaper/http/utils.rb:56:in `perform_request'
	from /Users/stanleystevens/.rvm/gems/ruby-2.2.1/gems/instapaper-1.0.0.pre2/lib/instapaper/http/utils.rb:52:in `perform_post_with_unparsed_response'
	from /Users/stanleystevens/.rvm/gems/ruby-2.2.1/gems/instapaper-1.0.0.pre2/lib/instapaper/api/oauth.rb:10:in `access_token'
	from login.rb:32:in `<main>'

and this is my code:

credentials = {
  consumer_key: 'foobarfoo',
  consumer_secret: 'barfoobar',
}

client = Instapaper::Client.new(credentials)
token = client.access_token("foo", "bar")

client.oauth_token = token.oauth_token
client.oauth_token_secret = token.oauth_token_secret

client.verify_credentials

I know all my creds are correct because it worked with the previous gem. Any help would be appreciated, thanks Steve!

**Comment by stve** [comment]
hi @Stanleyyork can you post a gist with your `Gemfile` and `Gemfile.lock` (or a scaled down Gemfile with just the basics. I can't seem to reproduce this issue.

**Comment by stve** [comment]
Closing this one out as it's quite old. I think it had to do with the version of http gem that was used. I've just released `1.0.0` of this gem which used http ~> 2.x. Hopefully these issues have gone away. If not, feel free to reopen.
